src/configuration.ts

```diff
@@ -0,0 +1,89 @@
+import { Environment, FormatEnum, PrecomputedFlag } from './interfaces';
+import { generateSalt, obfuscatePrecomputedFlags, Salt } from './obfuscation';
+import { Attributes, ContextAttributes } from './types';
+
+export interface PrecomputedConfigurationResponse {
+  // `format` is always `PRECOMPUTED`
+  readonly format: FormatEnum;
+  readonly obfuscated: boolean;
+  // Salt used for hashing md5-encoded strings.
+  readonly salt: string;
+  readonly createdAt: string;
+  // Environment might be missing if configuration was absent during evaluation.
+  readonly environment?: Environment;
+  readonly flags: Record<string, PrecomputedFlag>;
+}
+
+export interface IPrecomputedConfiguration {
+  // JSON encoded `PrecomputedConfigurationResponse`
+  readonly response: string;
+  readonly subjectKey: string;
+  // Optional in case server does not want to expose attributes to a client.
+  readonly subjectAttributes?: Attributes | ContextAttributes;
+}
+
+export class PrecomputedConfiguration implements IPrecomputedConfiguration {
+  readonly format = FormatEnum.PRECOMPUTED;
+  readonly response: string;
+
+  constructor(
+    readonly subjectKey: string,
+    flags: Record<string, PrecomputedFlag>,
+    readonly subjectAttributes?: Attributes | ContextAttributes,
+    environment?: Environment,
+  ) {
+    const precomputedResponse: PrecomputedConfigurationResponse = {
+      format: FormatEnum.PRECOMPUTED,
+      obfuscated: false,
+      salt: '',
+      createdAt: new Date().toISOString(),
+      environment,
+      flags,
+    };
+    this.response = JSON.stringify(precomputedResponse);
+  }
+}
+
+export class ObfuscatedPrecomputedConfiguration implements IPrecomputedConfiguration {
+  readonly format = FormatEnum.PRECOMPUTED;
+  readonly response: string;
+  private saltBase: Salt;
+
+  constructor(
+    readonly subjectKey: string,
+    flags: Record<string, PrecomputedFlag>,
+    readonly subjectAttributes?: Attributes | ContextAttributes,
+    environment?: Environment,
+  ) {
+    this.saltBase = generateSalt();
+
+    const precomputedResponse: PrecomputedConfigurationResponse = {
+      format: FormatEnum.PRECOMPUTED,
+      obfuscated: true,
+      salt: this.saltBase.base64String,
+      createdAt: new Date().toISOString(),
+      environment,
+      flags: obfuscatePrecomputedFlags(this.saltBase.saltString, flags),
+    };
+    this.response = JSON.stringify(precomputedResponse);
+  }
+}
+
+// "Wire" in the name means "in-transit"/"file" format.
+// In-memory representation may differ significantly and is up to SDKs.
+export interface ConfigurationWire {
```

Comment:
Can you include this in the exported interfaces in `index.ts`?

https://github.com/Eppo-exp/js-sdk-common/blob/cc921efeb10cec564850b3963de7b40dbfb6325e/src/index.ts#L94-L104

Then I can access this in the JS client SDK to define the expected input type to offline init the client

Reply:
Done